Formatting Changes & GUI Instructions

balajsra opened the pull request:
- Add formatting changes (create numbered and unordered lists and sub-lists)
- Add more Docker vocab
- Add information for each step of ROS tutorial

balajsra committed:
Lists and ROS Tutorial Info

- Add bullet point and numbered lists to improve readability
- Add more ROS vocabulary
- Add helpful tips for ROS Tutorial

balajsra committed:
Docker GUI Instructions

- Add docker GUI instructions for Windows
- Add commands to set source and run tasks in background

balajsra committed:
More GUI Info & Auto-Install ROS Tutorial Files

- Add example IP Address to docker-compose.yml for GUI
- Update README with step-by-step guide for enabling GUIs on Docker
- Add note to run XLaunch before running container for GUIs in README
- Add tip for running process in background
- Update Dockerfile to install ros-tutorials files on build
- Remove roscore from initial ROS Tutorial Steps

balajsra committed:
Add docker-compose comment

- Add comment to docker-compose indicating that IP Address needs to be changed
- Update docker-compose.yml example in README with comment

balajsra committed:
Tmux instructions and Dockerfile Update

- Add rqt package installs to Dockerfile
- Add information on using WSL and tmux in docker to README

balajsra committed:
Minor Fixes

- Fix tmux steps: run docker first, then tmux
- Moved ROS vocab to a new section
- Add note to manually run docker VM if using WSL
- Add summary of ROS Commands from tutorials

rywunder commented on README.md:
```diff
@@ -3,17 +3,72 @@
+    - `sudo apt-get install tmux`
+4. Follow the steps in ROS Tutorial below to run the docker container
+5. When inside docker, open tmux and run roscore in one window
+    - `tmux`
```
As an FYI this is a great post on making TMUX more usable 

https://www.hamvocke.com/blog/a-guide-to-customizing-your-tmux-conf/

Then you just have to mount the .tmux.conf file to use the changes your docker container

rywunder replied:
Also it might be useful to mention the other way to launch a new terminal. `docker exec -it <container_name> bash`

This would avoid people having to learn a new program. Although tmux is the best

balajsra replied:
Implemented in **tmux Customization and docker exec for new windows** commit

balajsra committed:
tmux Customization and docker exec for new windows

- Minor name change to Suggested Improvement
- Add link to article discussion tmux customization
- Add new section for running a new docker terminal window without tmux
